packages/pharos/src/components/toast/pharos-toaster.ts

```diff
@@ -52,45 +103,72 @@ export class PharosToaster extends PharosElement {
     super.disconnectedCallback && super.disconnectedCallback();
   }
 
-  private _getToastID(id: string | null) {
+  private _getToastID(id: ToastID | null | undefined) {
     return id || `toast_${uuidv4()}`;
   }
 
   private async _openToast(event: Event): Promise<void> {
-    const toastTag = this.localName.split('pharos-toaster')[0] + 'pharos-toast';
-    const toast = document.createElement(toastTag) as PharosToast;
-    const { content, status, id, indefinite } = (<CustomEvent>event).detail;
-
-    toast.innerHTML = content;
-    toast.status = status || DEFAULT_STATUS;
-    toast.id = this._getToastID(id);
-    toast.indefinite = indefinite || DEFAULT_INDEFINITE;
-    this.insertBefore(toast, this.childNodes[0] || null);
+    const { content, status, id, indefinite } = <ToastCreateDetail>(<CustomEvent>event).detail;
+    const toastId = this._getToastID(id);
+
+    this._toasts = [
+      {
+        content,
+        status: status || DEFAULT_STATUS,
+        id: toastId,
+        indefinite: indefinite || DEFAULT_INDEFINITE,
+      },
+      ...this._toasts,
+    ];
+
     await this.updateComplete;
-    toast.focus();
+    (this.renderRoot.querySelector(`#${toastId}`) as HTMLElement)?.focus();
   }
 
   private _updateToast(event: CustomEvent): void {
-    const { content, status, id } = (<CustomEvent>event).detail;
-    const toast = document.getElementById(this._getToastID(id));
-    if (toast) {
-      toast.innerHTML = content;
-      toast.status = status;
-    }
+    const { content, status, id } = <ToastUpdateDetail>(<CustomEvent>event).detail;
+
+    this._toasts = this._toasts.map((toast: ToastDetail): ToastDetail => {
+      if (toast.id === id) {
+        return {
+          ...toast,
+          content: content || toast.content,
+          status: status || toast.status,
+        };
+      } else {
+        return toast;
+      }
+    });
   }
 
   private _closeToast(event: CustomEvent): void {
-    const { id } = (<CustomEvent>event).detail || {};
-    const toast = document.getElementById(this._getToastID(id));
-    if (toast) {
-      this.removeChild(toast);
-    }
+    const { id } = <ToastCloseDetail>(<CustomEvent>event).detail || {};
+    this._toasts = this._toasts.filter((toast) => toast.id !== id);
+  }
+
+  private _renderToast(toast: ToastDetail): TemplateResult {
+    // This is used to properly render any Pharos components present in the supplied content.
+    // unsafeHtml will render content in general, but does not appear to render components.
+    const toastContentElement = document.createElement('div');
+    toastContentElement.innerHTML = toast.content;
+
+    return html`<pharos-toast
+      id="${toast.id}"
+      status="${toast.status}"
+      ?indefinite="${toast.indefinite}"
+    >
+      ${toastContentElement}
+    </pharos-toast>`;
```

Comment:
@Niznikr curious if you might understand more about why this is and if I've missed some trick here.